src/arch/ia32e/address.rs

```diff
@@ -0,0 +1,84 @@
+pub struct VirtualAddress(u64);
+
+impl VirtualAddress {
+    /// Try to convert u64 into an ia32e virtual address.
+    ///
+    /// # Safety
+    ///
+    /// This function ensures the virtual address is correct.
+    pub fn try_new(addr: u64) -> Option<Self>
+    {
+        if addr < (1 << 48) {
+            Some(VirtualAddress(addr))
+        } else {
+            None
+        }
+    }
+
+    /// Convert u64 into an ia32e virtual address.
+    ///
+    /// # Safety
+    ///
+    /// This function panics if the virtual address is incorrect.
+    pub fn new(addr: u64) -> Self
+    {
+        Self::try_new(addr).expect("Virtual address cannot be longer than 48 bits.")
+    }
+}
```

Comment:
We may also require a function without any check. Something like `new_unchecked` with no verification whatsoever of the given data. You may need to tag this function as unsafe.